### DISCUSSION TITLE
Logical operators (or/and) in GraphQL where parameter

### DISCUSSION THREAD

**Discussion by skourismanolis** [strapi/strapi#7795] [Legacy Topics]
Hi, I couldn't find a way to use "and" and "or" operators in graphql. As I understand from #3194 the functionality exists, but I couldn't find any documentation. What I'm trying to do is something like:
```gql

  people(
    where: {
      OR: [
        { first_name_contains: "something" }
        { last_name_contains: "something else" }
      ]
    }
  ) {
    id
    first_name
    last_name
  }
}
```

I also tried using `$or`,`or` and `$OR` in place of `OR` but no dice.

**Comment by derrickmehaffy** [comment]
@alexandrebodin I believe you had a way to handle this with the new complex queries but I can't seem to find the reference. Can you describe how we use those and I can try to write documentation for it?

**Reply by alexandrebodin** [reply]
rename the OR with _or. the syntax inside of where is the same as the one described in the parameters doc here: https://strapi.io/documentation/v3.x/content-api/parameters.html#complex-queries.

**Reply by skourismanolis** [reply]
Thanks a lot! This worked, I initially read that part but since it was referring to REST filtering and I found the resulting query string quite complex I didn't think of applying it to GraphQL. It would be amazing if you wrote documentation for it @derrickmehaffy so it can be clearer.

**Reply by wujun4code** [reply]
@aalasolutions how abount `AND` ?
i try the following query, it did not work:

```
query {
  products(where:{ _and: [ { labels: { name:"category",value: "blade" }},
    { labels: { name:"category",value: "featured" }}]}) {
    name
    unique_name
    labels {
      name
    }
  }
}

```

Thanks,

**Reply by wujun4code** [reply]
my related issue is from https://stackoverflow.com/questions/41515226/graphql-filter-data-in-an-array, i use a category like a tag for product.

**Reply by wujun4code** [reply]
my rest api request is : 
```
curl --location --request GET 'localhost:1337/products?_where[0][labels.name]=category&_where[1][labels.value]=blade&_where[2][labels.name]=category&_where[3][labels.value]=featured'
```
it works well.

**Reply by alexandrebodin** [reply]
THere are no and this is implicit in the _where

You can add 
```graphql
query {
  products(where:{ _where: [ { labels: { name:"category",value: "blade" }},
    { labels: { name:"category",value: "featured" }}]}) {
    name
    unique_name
    labels {
      name
    }
  }
}

```

**Reply by wujun4code** [reply]
@alexandrebodin but i got the following error:

```
{
    "errors": [
        {
            "message": "Cast to ObjectId failed for value \"{ name: 'category', value: 'pupular-items' }\" at path \"labels\" for model \"product\"",
            "locations": [
                {
                    "line": 2,
                    "column": 3
                }
            ],
            "path": [
                "products"
            ],
            "extensions": {
                "code": "INTERNAL_SERVER_ERROR",
                "exception": {
                    "stringValue": "\"{ name: 'category', value: 'pupular-items' }\"",
                    "kind": "ObjectId",
                    "value": {
                        "name": "category",
                        "value": "pupular-items"
                    },
                    "path": "labels",
                    "reason": {}
                }
            }
        }
    ],
    "data": {
        "products": null
    }
}
```

**Comment by derrickmehaffy** [comment]
Locking as migrated to: https://forum.strapi.io/t/logical-operators-or-and-in-graphql-where-parameter-7795/1753
